From a7d69bbd83ee47b0af4276e0203c0de6623ab2ee Mon Sep 17 00:00:00 2001
Message-Id: <a7d69bbd83ee47b0af4276e0203c0de6623ab2ee.1524081065.git.Jim.Somerville@windriver.com>
In-Reply-To: <8d084dcc717666af0283e4ddeefaba7e331d879a.1524081065.git.Jim.Somerville@windriver.com>
References: <8d084dcc717666af0283e4ddeefaba7e331d879a.1524081065.git.Jim.Somerville@windriver.com>
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Tue, 18 Jul 2017 15:52:25 -0400
Subject: [PATCH 20/39] Fix stack clash

Consists of 3 patches taken from the linux-3.10.y branch of the
linux-stable tree.

The first (main) patch applied fairly cleanly with offsets,
requiring a small amount of wiggle and a bit of manual work
due to some affected code residing in mm/gup.c .
The other 2 patches applied cleanly.

This fix here is only temporary and won't be needed once we
move to a newer released CentOS kernel that includes it.

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 SPECS/kernel-rt.spec | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/SPECS/kernel-rt.spec b/SPECS/kernel-rt.spec
index 0bfaf4e..195db97 100644
--- a/SPECS/kernel-rt.spec
+++ b/SPECS/kernel-rt.spec
@@ -410,6 +410,9 @@ Patch1015: Enable-building-kernel-with-CONFIG_BLK_DEV_NBD.patch
 Patch1016: x86-make-dma_alloc_coherent-return-zeroed-memory-if-.patch
 Patch1017: arch-x86-kernel-pci-dma.c-fix-dma_generic_alloc_cohe.patch
 Patch1018: rcu-Don-t-wake-rcuc-X-kthreads-on-NOCB-CPUs.patch
+Patch1019: mm-larger-stack-guard-gap-between-vmas.patch
+Patch1020: mm-fix-new-crash-in-unmapped_area_topdown.patch
+Patch1021: Allow-stack-to-grow-up-to-address-space-limit.patch
 
 BuildRoot: %{_tmppath}/kernel-%{KVERREL}-root
 
@@ -788,6 +791,9 @@ ApplyPatch Enable-building-kernel-with-CONFIG_BLK_DEV_NBD.patch
 ApplyPatch x86-make-dma_alloc_coherent-return-zeroed-memory-if-.patch
 ApplyPatch arch-x86-kernel-pci-dma.c-fix-dma_generic_alloc_cohe.patch
 ApplyPatch rcu-Don-t-wake-rcuc-X-kthreads-on-NOCB-CPUs.patch
+ApplyPatch mm-larger-stack-guard-gap-between-vmas.patch
+ApplyPatch mm-fix-new-crash-in-unmapped_area_topdown.patch
+ApplyPatch Allow-stack-to-grow-up-to-address-space-limit.patch
 
 # move off upstream version mechanism
 if [ -e localversion-rt ]; then
-- 
1.8.3.1

